### ISSUE TITLE
Each time I restore Home Assistant core, I see the following Supervisor warning:  "WARNING (MainThread) [supervisor.homeassistant.core] Watchdog found Home Assistant failed, restarting..."

### ISSUE THREAD

**Issue by mkanet** [home-assistant/supervisor#4516]
### Describe the issue you are experiencing

I have a Home Assistant VMWare VM.  If I restart Home Assistant Core via Home Assistant UI (or when I restart the entire VM via VMWare Hypervisor), I see the following warning message in my supervisor log after each restart:

```
WARNING (MainThread) [supervisor.homeassistant.core] Watchdog found Home Assistant failed, restarting...
```

Could someone please shed some light on this issue.  I'm not sure why this is happening and how to avoid it.

### What type of installation are you running?
Home Assistant OS Version 10.5

### Anything in the Supervisor logs that might be useful for us?
```txt
23-08-31 13:04:35 INFO (MainThread) [supervisor.host.apparmor] Loading AppArmor Profiles: {'hassio-supervisor'}
23-08-31 13:04:35 INFO (MainThread) [supervisor.docker.monitor] Started docker events monitor
23-08-31 13:04:35 INFO (MainThread) [supervisor.docker.interface] Attaching to ghcr.io/home-assistant/amd64-hassio-cli with version 2023.06.0
23-08-31 13:04:35 INFO (MainThread) [supervisor.plugins.cli] Starting CLI plugin
23-08-31 13:04:35 INFO (SyncWorker_0) [supervisor.docker.manager] Cleaning hassio_cli application
23-08-31 13:04:35 INFO (MainThread) [supervisor.docker.cli] Starting CLI ghcr.io/home-assistant/amd64-hassio-cli with version 2023.06.0 - 172.30.32.5
23-08-31 13:04:35 INFO (MainThread) [supervisor.docker.interface] Attaching to ghcr.io/home-assistant/amd64-hassio-dns with version 2023.06.2
23-08-31 13:04:35 INFO (MainThread) [supervisor.plugins.dns] Starting CoreDNS plugin
23-08-31 13:04:35 INFO (SyncWorker_0) [supervisor.docker.manager] Cleaning hassio_dns application
23-08-31 13:04:35 INFO (MainThread) [supervisor.docker.dns] Starting DNS ghcr.io/home-assistant/amd64-hassio-dns with version 2023.06.2 - 172.30.32.3
23-08-31 13:04:35 INFO (MainThread) [supervisor.plugins.dns] Updated /etc/resolv.conf
23-08-31 13:04:35 INFO (MainThread) [supervisor.docker.interface] Attaching to ghcr.io/home-assistant/amd64-hassio-audio with version 2023.06.0
23-08-31 13:04:35 INFO (MainThread) [supervisor.plugins.audio] Starting Audio plugin
23-08-31 13:04:35 INFO (SyncWorker_1) [supervisor.docker.manager] Cleaning hassio_audio application
23-08-31 13:04:36 INFO (MainThread) [supervisor.docker.audio] Starting Audio ghcr.io/home-assistant/amd64-hassio-audio with version 2023.06.0 - 172.30.32.4
23-08-31 13:04:36 INFO (MainThread) [supervisor.docker.interface] Attaching to ghcr.io/home-assistant/amd64-hassio-observer with version 2023.06.0
23-08-31 13:04:36 INFO (MainThread) [supervisor.docker.interface] Attaching to ghcr.io/home-assistant/amd64-hassio-multicast with version 2023.06.2
23-08-31 13:04:36 INFO (MainThread) [supervisor.plugins.multicast] Starting Multicast plugin
23-08-31 13:04:36 INFO (SyncWorker_1) [supervisor.docker.manager] Cleaning hassio_multicast application
23-08-31 13:04:36 INFO (MainThread) [supervisor.docker.multicast] Starting Multicast ghcr.io/home-assistant/amd64-hassio-multicast with version 2023.06.2 - Host
23-08-31 13:04:36 INFO (MainThread) [supervisor.updater] Fetching update data from https://version.home-assistant.io/stable.json
23-08-31 13:04:36 INFO (MainThread) [supervisor.homeassistant.secrets] Loaded 1 Home Assistant secrets
23-08-31 13:04:36 INFO (MainThread) [supervisor.docker.interface] Attaching to ghcr.io/home-assistant/qemux86-64-homeassistant with version 2023.8.4

23-08-31 13:04:36 WARNING (MainThread) [supervisor.homeassistant.core] Watchdog found Home Assistant failed, restarting...

23-08-31 13:04:36 INFO (MainThread) [supervisor.os.manager] Detect Home Assistant Operating System 10.5 / BootSlot A
23-08-31 13:04:36 INFO (SyncWorker_1) [supervisor.docker.manager] Starting homeassistant
23-08-31 13:04:36 INFO (MainThread) [supervisor.homeassistant.core] Wait until Home Assistant is ready
23-08-31 13:04:36 INFO (MainThread) [supervisor.store.git] Loading add-on /data/addons/git/a0d7b954 repository
23-08-31 13:04:36 INFO (MainThread) [supervisor.store.git] Loading add-on /data/addons/git/5c53de3b repository
23-08-31 13:04:36 INFO (MainThread) [supervisor.store.git] Loading add-on /data/addons/core repository
23-08-31 13:04:37 INFO (MainThread) [supervisor.store] Loading add-ons from store: 71 all - 71 new - 0 remove
23-08-31 13:04:37 INFO (MainThread) [supervisor.addons] Found 5 installed add-ons
23-08-31 13:04:37 INFO (MainThread) [supervisor.docker.interface] Attaching to homeassistant/amd64-addon-mosquitto with version 6.2.1
23-08-31 13:04:37 INFO (MainThread) [supervisor.docker.interface] Attaching to homeassistant/amd64-addon-samba with version 10.0.2
23-08-31 13:04:37 INFO (MainThread) [supervisor.docker.interface] Attaching to homeassistant/amd64-addon-configurator with version 5.6.0
23-08-31 13:04:37 INFO (MainThread) [supervisor.docker.interface] Attaching to homeassistant/amd64-addon-zwave_js with version 0.1.88
23-08-31 13:04:37 INFO (MainThread) [supervisor.docker.interface] Attaching to homeassistant/amd64-addon-ssh with version 9.7.1
23-08-31 13:04:37 INFO (MainThread) [supervisor.backups.manager] Found 5 backup files
23-08-31 13:04:37 INFO (MainThread) [supervisor.discovery] Loaded 2 messages
23-08-31 13:04:37 INFO (MainThread) [supervisor.ingress] Loaded 0 ingress sessions
23-08-31 13:04:37 INFO (MainThread) [supervisor.resolution.check] Starting system checks with state CoreState.SETUP
23-08-31 13:04:37 INFO (MainThread) [supervisor.resolution.check] System checks complete
23-08-31 13:04:37 INFO (MainThread) [supervisor.resolution.evaluate] Starting system evaluation with state CoreState.SETUP
23-08-31 13:04:37 INFO (MainThread) [supervisor.resolution.evaluate] System evaluation complete
23-08-31 13:04:37 INFO (MainThread) [supervisor.jobs] 'ResolutionFixup.run_autofix' blocked from execution, system is not running - CoreState.SETUP
23-08-31 13:04:37 INFO (MainThread) [supervisor.resolution.evaluate] Starting system evaluation with state CoreState.SETUP
23-08-31 13:04:37 INFO (MainThread) [supervisor.resolution.evaluate] System evaluation complete
23-08-31 13:04:37 INFO (MainThread) [__main__] Running Supervisor
23-08-31 13:04:37 INFO (MainThread) [supervisor.os.manager] Rauc: A - marked slot kernel.0 as good
23-08-31 13:04:37 INFO (MainThread) [supervisor.addons] Phase 'AddonStartup.INITIALIZE' starting 0 add-ons
23-08-31 13:04:37 INFO (MainThread) [supervisor.addons] Phase 'AddonStartup.SYSTEM' starting 1 add-ons
23-08-31 13:04:37 INFO (SyncWorker_3) [supervisor.docker.manager] Cleaning addon_core_mosquitto application
23-08-31 13:04:37 INFO (MainThread) [supervisor.docker.addon] Starting Docker add-on homeassistant/amd64-addon-mosquitto with version 6.2.1
23-08-31 13:04:38 INFO (MainThread) [supervisor.resolution.evaluate] Starting system evaluation with state CoreState.STARTUP
23-08-31 13:04:38 INFO (MainThread) [supervisor.resolution.evaluate] System evaluation complete
23-08-31 13:04:39 INFO (MainThread) [supervisor.store.git] Update add-on https://github.com/esphome/home-assistant-addon repository
23-08-31 13:04:39 INFO (MainThread) [supervisor.store.git] Update add-on https://github.com/hassio-addons/repository repository
23-08-31 13:04:39 INFO (MainThread) [supervisor.store.git] Update add-on https://github.com/home-assistant/addons repository
23-08-31 13:04:39 INFO (MainThread) [supervisor.services.modules.mqtt] Set core_mosquitto as service provider for mqtt
23-08-31 13:04:40 INFO (MainThread) [supervisor.store] Loading add-ons from store: 71 all - 0 new - 0 remove
23-08-31 13:04:40 INFO (MainThread) [supervisor.store] Loading add-ons from store: 71 all - 0 new - 0 remove
23-08-31 13:04:41 INFO (MainThread) [supervisor.homeassistant.api] Updated Home Assistant API token
23-08-31 13:04:42 INFO (MainThread) [supervisor.addons] Phase 'AddonStartup.SERVICES' starting 2 add-ons
23-08-31 13:04:42 INFO (SyncWorker_2) [supervisor.docker.manager] Cleaning addon_core_ssh application
23-08-31 13:04:43 INFO (MainThread) [supervisor.docker.addon] Starting Docker add-on homeassistant/amd64-addon-ssh with version 9.7.1
23-08-31 13:04:43 INFO (SyncWorker_0) [supervisor.docker.manager] Cleaning addon_core_samba application
23-08-31 13:04:43 INFO (MainThread) [supervisor.docker.addon] Starting Docker add-on homeassistant/amd64-addon-samba with version 10.0.2
23-08-31 13:04:43 INFO (MainThread) [supervisor.api.middleware.security] /network/info access from core_samba
23-08-31 13:04:46 INFO (SyncWorker_3) [supervisor.docker.manager] Cleaning addon_core_zwave_js application
23-08-31 13:04:46 INFO (MainThread) [supervisor.docker.addon] Starting Docker add-on homeassistant/amd64-addon-zwave_js with version 0.1.88
23-08-31 13:04:47 INFO (MainThread) [supervisor.api.middleware.security] /host/info access from core_zwave_js
23-08-31 13:04:48 INFO (MainThread) [supervisor.core] Skiping start of Home Assistant
23-08-31 13:04:48 INFO (MainThread) [supervisor.addons] Phase 'AddonStartup.APPLICATION' starting 1 add-ons
23-08-31 13:04:48 INFO (SyncWorker_3) [supervisor.docker.manager] Cleaning addon_core_configurator application
23-08-31 13:04:48 INFO (MainThread) [supervisor.docker.addon] Starting Docker add-on homeassistant/amd64-addon-configurator with version 5.6.0
23-08-31 13:05:01 INFO (MainThread) [supervisor.homeassistant.core] Detect a running Home Assistant instance
23-08-31 13:05:19 INFO (MainThread) [supervisor.misc.tasks] All core tasks are scheduled
23-08-31 13:05:19 INFO (MainThread) [supervisor.core] Supervisor is up and running
23-08-31 13:05:19 INFO (MainThread) [supervisor.host.info] Updating local host information
23-08-31 13:05:19 INFO (MainThread) [supervisor.updater] Fetching update data from https://version.home-assistant.io/stable.json
23-08-31 13:05:19 INFO (MainThread) [supervisor.resolution.check] Starting system checks with state CoreState.RUNNING
23-08-31 13:05:19 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.FREE_SPACE/ContextType.SYSTEM
23-08-31 13:05:19 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.TRUST/ContextType.SUPERVISOR
23-08-31 13:05:19 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.MULTIPLE_DATA_DISKS/ContextType.SYSTEM
23-08-31 13:05:19 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.SECURITY/ContextType.CORE
23-08-31 13:05:19 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.DNS_SERVER_FAILED/ContextType.DNS_SERVER
23-08-31 13:05:19 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.DNS_SERVER_IPV6_ERROR/ContextType.DNS_SERVER
23-08-31 13:05:19 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.DOCKER_CONFIG/ContextType.SYSTEM
23-08-31 13:05:19 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.PWNED/ContextType.ADDON
23-08-31 13:05:19 INFO (MainThread) [supervisor.host.services] Updating service information
23-08-31 13:05:19 INFO (MainThread) [supervisor.host.network] Updating local network information
23-08-31 13:05:19 INFO (MainThread) [supervisor.host.sound] Updating PulseAudio information
23-08-31 13:05:19 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.IPV4_CONNECTION_PROBLEM/ContextType.SYSTEM
23-08-31 13:05:19 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.NO_CURRENT_BACKUP/ContextType.SYSTEM
23-08-31 13:05:19 INFO (MainThread) [supervisor.resolution.check] System checks complete
23-08-31 13:05:19 INFO (MainThread) [supervisor.resolution.evaluate] Starting system evaluation with state CoreState.RUNNING
23-08-31 13:05:19 INFO (MainThread) [supervisor.host.manager] Host information reload completed
23-08-31 13:05:19 INFO (MainThread) [supervisor.resolution.evaluate] System evaluation complete
23-08-31 13:05:19 INFO (MainThread) [supervisor.resolution.fixup] Starting system autofix at state CoreState.RUNNING
23-08-31 13:05:19 INFO (MainThread) [supervisor.resolution.fixup] System autofix complete
```

## System Information

version | core-2023.8.4
-- | --
installation_type | Home Assistant OS
dev | false
hassio | true
docker | true
user | root
virtualenv | false
python_version | 3.11.4
os_name | Linux
os_version | 6.1.45
arch | x86_64
timezone | America/Los_Angeles
config_dir | /config

<details><summary>Home Assistant Community Store</summary>

GitHub API | ok
-- | --
GitHub Content | ok
GitHub Web | ok
GitHub API Calls Remaining | 4730
Installed Version | 1.32.1
Stage | running
Available Repositories | 1277
Downloaded Repositories | 14

</details>

<details><summary>Home Assistant Cloud</summary>

logged_in | true
-- | --
subscription_expiration | July 28, 2024 at 5:00 PM
relayer_connected | true
relayer_region | us-east-1
remote_enabled | false
remote_connected | false
alexa_enabled | true
google_enabled | true
remote_server | us-east-1-6.ui.nabu.casa
certificate_status | ready
can_reach_cert_server | ok
can_reach_cloud_auth | ok
can_reach_cloud | ok

</details>

<details><summary>Home Assistant Supervisor</summary>

host_os | Home Assistant OS 10.5
-- | --
update_channel | stable
supervisor_version | supervisor-2023.08.1
agent_version | 1.5.1
docker_version | 23.0.6
disk_total | 30.8 GB
disk_used | 4.9 GB
healthy | true
supported | true
board | ova
supervisor_api | ok
version_api | ok
installed_addons | Terminal & SSH (9.7.1), Samba share (10.0.2), Mosquitto broker (6.2.1), File editor (5.6.0), Z-Wave JS (0.1.88)

</details>

<details><summary>Dashboards</summary>

dashboards | 1
-- | --
resources | 14
views | 7
mode | storage

</details>

<details><summary>Recorder</summary>

oldest_recorder_run | August 24, 2023 at 8:13 PM
-- | --
current_recorder_run | August 31, 2023 at 1:04 PM
estimated_db_size | 345.45 MiB
database_engine | sqlite
database_version | 3.41.2

</details>

### Supervisor diagnostics

[config_entry-hassio-c949225816e9f21c12b25db60074d4f5.json (1).txt](https://github.com/home-assistant/supervisor/files/12489976/config_entry-hassio-c949225816e9f21c12b25db60074d4f5.json.1.txt)

**Comment by mkanet** [comment]
Update, it looks like restarting Home Assistant Core from Home Assistant also produces the log warning "WARNING (MainThread) [supervisor.homeassistant.core] Watchdog found Home Assistant failed, restarting..." after I restart HA Core".   This happens every single time.

**Comment by mdegat01** [comment]
Do you still see this? I just tried to make and restore a backup and I can't seem to reproduce it. Here's my supervisor logs during this:
```
23-09-08 11:02:14 INFO (MainThread) [supervisor.backups.manager] Backup f35d6efa starting stage addon_repositories
23-09-08 11:02:14 INFO (MainThread) [supervisor.backups.manager] Backup f35d6efa starting stage docker_config
23-09-08 11:02:14 INFO (MainThread) [supervisor.backups.manager] Creating new partial backup with slug f35d6efa
23-09-08 11:02:14 INFO (MainThread) [supervisor.backups.manager] Backup f35d6efa starting stage home_assistant
23-09-08 11:02:14 INFO (MainThread) [supervisor.homeassistant.module] Backing up Home Assistant Core config folder
23-09-08 11:02:14 INFO (MainThread) [supervisor.homeassistant.module] Backup Home Assistant Core config folder done
23-09-08 11:02:14 INFO (MainThread) [supervisor.backups.manager] Backup f35d6efa starting stage finishing_file
23-09-08 11:02:14 INFO (MainThread) [supervisor.backups.manager] Creating partial backup with slug f35d6efa completed
23-09-08 11:02:33 INFO (MainThread) [supervisor.backups.manager] Partial-Restore f35d6efa start
23-09-08 11:02:33 INFO (MainThread) [supervisor.backups.manager] Restore f35d6efa starting stage docker_config
23-09-08 11:02:33 INFO (MainThread) [supervisor.backups.manager] Restore f35d6efa starting stage home_assistant
23-09-08 11:02:33 INFO (SyncWorker_0) [supervisor.docker.manager] Stopping homeassistant application
23-09-08 11:03:07 INFO (MainThread) [supervisor.homeassistant.module] Restore Home Assistant Core config folder
23-09-08 11:03:07 INFO (MainThread) [asyncio] execute program 'bash': <_UnixSubprocessTransport pid=7907 running>
23-09-08 11:03:07 INFO (MainThread) [asyncio] <_UnixSubprocessTransport pid=7907 running> exited with return code 0
23-09-08 11:03:07 INFO (MainThread) [supervisor.homeassistant.module] Restore Home Assistant Core config folder done
23-09-08 11:03:07 INFO (MainThread) [supervisor.homeassistant.module] Restore Home Assistant Core metadata
23-09-08 11:03:07 INFO (MainThread) [supervisor.backups.manager] Restore f35d6efa starting stage await_home_assistant_restart
23-09-08 11:03:07 INFO (MainThread) [supervisor.backups.manager] Restore f35d6efa starting stage check_home_assistant
23-09-08 11:03:07 INFO (SyncWorker_3) [supervisor.docker.manager] Starting homeassistant
23-09-08 11:03:08 INFO (MainThread) [supervisor.homeassistant.core] Wait until Home Assistant is ready
23-09-08 11:03:11 INFO (MainThread) [supervisor.resolution.evaluate] Starting system evaluation with state freeze
23-09-08 11:03:11 INFO (MainThread) [supervisor.resolution.evaluate] System evaluation complete
23-09-08 11:03:14 INFO (MainThread) [supervisor.updater] Fetching update data from https://version.home-assistant.io/dev.json
23-09-08 11:03:14 WARNING (MainThread) [supervisor.jobs] 'StoreManager.reload' blocked from execution, supervisor needs to be updated first
23-09-08 11:03:44 INFO (MainThread) [supervisor.homeassistant.core] Detect a running Home Assistant instance
23-09-08 11:03:44 INFO (MainThread) [supervisor.backups.manager] Partial-Restore f35d6efa done
```
I see you did this on supervisor `2023.08.1`. There was a bug fix related to backup restore not properly stopping things that went out in `2023.08.3` (the current stable version of supervisor). Wondering if that fixed this or if there is still something else going on here.

**Comment by mkanet** [comment]
@mdegat01 thank you for your reply.  I just updated to Supervosir 2023.08.1  The issue I was seeing happened immediately after rebooting Home Assistant.  However, it only happened once in a while... maybe it was after a subsequent backup.  I'm not certain.

I will post back here if I see a reoccurrence of this issue now that I upgraded Supervisor.

**Comment by mkanet** [comment]
@mdegat01 I am 90% sure this issue was introduced when I added the unsupported custom integration [Roblox](https://github.com/jdeath/Roblox-Homeassistant).  I noticed that when I add this custom component and restart Home Assistant, during the reboot, the WARNING message is introduced immediately.

I'm not sure what's happening.. but it seems like Home Assistant Core crashes/stop prematurely during a HA reboot.

At this point, I think the only thing I can do is remove that custom component.  Thank you for your time.
